docs/plugins.md

```diff
@@ -118,6 +118,7 @@ export const parsers = {
 The signature of the `parse` function is:
 
 ```ts
+function parse(text: string): Promise<AST> | AST;
```

Comment:
I think this signals what the arguments this function will receive, I don't think we should add this, many plugins define the following functions/methods with less parameters, we didn't mark they are optional. I'm not familiar with  TS, teach me if there is a good explanation.

Reply:
So, basically here I'm saying that there are TWO valid ways to call `parse`:

1. You can call it with a single argument, `parse(text)`, OR
2. You can call it with two arguments, `parse(text, options)`

Prior to this PR, the codebase can't make up its mind about whether the first option is supposed to be allowed or not. The [old declaration of the `Parser` interface](https://github.com/prettier/prettier/pull/15770/files#diff-619cc2239a7eb4ef05041767e6ab0e4fb069e98ee9d2c83c2d8e1865c475e141L450) said that the `options` argument was mandatory and that you cannot just call `parse(text)`. But *all* of our actual implementations of `parse` supported calling `parse(text)` one way or another (either they checked for options being null and handled that case, or they had a default argument of `{}`, both of which effectively mean you can call `parse(text)` without providing a second argument), and the unit test that I restore in this PR specifically tested that it's possible to call `parse(text)` without a second argument.

Clearly this inconsistency should be resolved one way or another. *Either* the previous types were wrong and needed changing to say that it's okay to call `parse(text)` (that's what I've assumed in this PR), *or* the test where we were calling `parse(text)` is wrong and should be calling `parse(text, options)` instead.

Reply:
`parse(test)` was added by me, but I forgot where it is...

Reply:
Found, https://github.com/prettier/prettier/issues/11888, we don't really support `parse(text)`.